### PR TITLE
feat: add gifsicle backend

### PR DESCRIPTION
**Pull request by yansal**

### CODE REVIEW

File: engine/backend/gifsicle.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+// Rotate implements Backend.
+func (b *Gifsicle) Rotate(*image.ImageFile, *Options) ([]byte, error) {
+	return nil, MethodNotImplementedError
+}
+
+// Thumbnail implements Backend.
+func (b *Gifsicle) Thumbnail(imgfile *image.ImageFile, opts *Options) ([]byte, error) {
+	cmd := exec.Command("gifsicle", "--resize", fmt.Sprintf("%dx%d", opts.Width, opts.Height))
```
Comment (thoas) [inline engine/backend/gifsicle.go:47]: `gifscile` should be the default command value but it needs to be customised in the config if the command is not in the user path which will be the case in production

File: engine/backend/gifsicle.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+// Flip implements Backend.
+func (b *Gifsicle) Flip(*image.ImageFile, *Options) ([]byte, error) {
+	return nil, MethodNotImplementedError
+}
+
+// Resize implements Backend.
+func (b *Gifsicle) Resize(*image.ImageFile, *Options) ([]byte, error) {
+	return nil, MethodNotImplementedError
```
Comment (thoas) [inline engine/backend/gifsicle.go:36]: we cannot implement it?

**Review by thoas — CHANGES_REQUESTED** [review]

File: engine/backend/gifsicle.go (hunk 2 above)
Comment (yansal) [reply engine/backend/gifsicle.go]: done

File: engine/backend/gifsicle.go (hunk 1 above)
Comment (yansal) [reply engine/backend/gifsicle.go]: done

**Comment by thoas** [comment]
great PR @yansal 💪

**Pull request merged**